employee-rostering-frontend/src/store/employee/selectors.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { AppState } from '../types';
+import { contractSelectors } from 'store/contract';
+import { skillSelectors } from 'store/skill';
+import Employee from 'domain/Employee';
+import DomainObjectView from 'domain/DomainObjectView';
+
+export const getEmployeeById = (state: AppState, id: number): Employee => {
+  if (state.employeeList.isLoading || state.skillList.isLoading || state.contractList.isLoading) {
+    throw Error("Employee list is loading");
+  }
+  const employeeView = state.employeeList.employeeMapById.get(id) as DomainObjectView<Employee>;
```

Comment:
Are you sure that `Map.get()` here never returns `undefined`? Is it guaranteed that an employee with the given `id` is always there in the map? Maybe it is guaranteed but it's not obvious from here and the Map API doesn't guarantee it so I'd probably rather see the `undefined`-check here instead of the cast. Wdyt?

Reply:
Since the `employeeList`, `skillList`, and `contractList` are finished loading (see the if statement above), all id's should be in the map. Moreover, since the id's come from other `DomainObjects` (with the same `tenantId`), the method would not be called with a non-existent employee id (unless someone modify the JavaScript variables in the console). Then we have the question of what to do with an id that not in the map; I would throw an Error (which is also the implicit current behaviour).

Reply:
Sounds good.